pallets/xyk/src/lib.rs

```diff
@@ -459,8 +460,17 @@ pub mod pallet {
 
 			let liquidity_a = T::Currency::total_balance(asset_a, &pair_account);
 			let liquidity_b = T::Currency::total_balance(asset_b, &pair_account);
-			T::AMMHandler::on_liquidity_changed(SOURCE, asset_a, asset_b, amount_a, amount_b, liquidity_a, liquidity_b)
-				.map_err(|(_w, e)| e)?;
+			T::AMMHandler::on_liquidity_changed(
+				SOURCE,
+				asset_a,
+				asset_b,
+				amount_a,
+				amount_b,
+				liquidity_a,
+				liquidity_b,
+				Ratio::new(liquidity_a, liquidity_b),
```

Comment:
@apopiak 

the price is correctly calculated ? I was not sure whether it should be amounts or liquidity.

Reply:
according to the oracle tests, the liquidity should be used here. But Alex might know better...

Reply:
Sorry I didn't see this in time. Yes it should be liquidity for XYK